### PR TITLE
fix: Ignore ComboBox client-side filter in data view items getters

### PR DESCRIPTION
**Pull request by mshabarov**
ComboBox's data views getItem() and getItems() methods implementations modified in a way to be not affected by client-side filter.

Fixes: https://github.com/vaadin/vaadin-flow-components/issues/282

### CODE REVIEW

File: vaadin-combo-box-flow-parent/vaadin-combo-box-flow/src/main/java/com/vaadin/flow/component/combobox/dataview/ItemFetchHelper.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+            throw new IndexOutOfBoundsException("Index must be non-negative");
+        }
+        if (dataCommunicator.isDefinedSize()) {
+            final int itemCount = dataCommunicator.getDataProviderSize();
+            if (itemCount == 0) {
+                throw new IndexOutOfBoundsException(String
+                        .format("Requested index %d on empty data.", index));
+            } else if (index >= itemCount) {
+                throw new IndexOutOfBoundsException(String.format(
+                        "Given index %d is outside of the accepted range '0 - %d'",
```
Comment (pleku) [inline vaadin-combo-box-flow-parent/vaadin-combo-box-flow/src/main/java/com/vaadin/flow/component/combobox/dataview/ItemFetchHelper.java:51]: Accepted range? What is that? Why not just say larger than the item count?

File: vaadin-combo-box-flow-parent/vaadin-combo-box-flow/src/main/java/com/vaadin/flow/component/combobox/dataview/ItemFetchHelper.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+package com.vaadin.flow.component.combobox.dataview;
+
+import java.io.Serializable;
+import java.util.stream.Stream;
+
+import com.vaadin.flow.data.provider.DataCommunicator;
+import com.vaadin.flow.data.provider.Query;
+
+/**
+ * Helper methods for fetching the ComboBox's items in data views.
```
Comment (pleku) [inline vaadin-combo-box-flow-parent/vaadin-combo-box-flow/src/main/java/com/vaadin/flow/component/combobox/dataview/ItemFetchHelper.java:26]: Nit: method javadocs don't hurt

**Review by pleku — CHANGES_REQUESTED** [review]
Small nits

File: vaadin-combo-box-flow-parent/vaadin-combo-box-flow/src/main/java/com/vaadin/flow/component/combobox/dataview/ItemFetchHelper.java (hunk 1 above)
Comment (mshabarov) [reply vaadin-combo-box-flow-parent/vaadin-combo-box-flow/src/main/java/com/vaadin/flow/component/combobox/dataview/ItemFetchHelper.java]: done

File: vaadin-combo-box-flow-parent/vaadin-combo-box-flow/src/main/java/com/vaadin/flow/component/combobox/dataview/ItemFetchHelper.java (hunk 2 above)
Comment (mshabarov) [reply vaadin-combo-box-flow-parent/vaadin-combo-box-flow/src/main/java/com/vaadin/flow/component/combobox/dataview/ItemFetchHelper.java:26]: done

**Review by pleku — APPROVED** [review]

**Pull request merged**